pandas/_libs/lib.pyx

```diff
@@ -2526,3 +2526,20 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def from_nested_dict(dict data) -> dict:
+    cdef:
+        dict new_data = {}
+        object index, column, value, dict_or_series
+
+    for index, dict_or_series in data.items():
+        for column, value in dict_or_series.items():
+            if column in new_data:
+                new_data[column].update(dict([(index, value)]))
```

Comment:
The ```dict([(index, value)])``` seems like I am doing an unneeded round trip, but I could not found another way to do this, any ideas?

Reply:
Maybe just do the loop more explicit:

```python
for index, dict_or_series in data.items():
    for column, value in dict_or_series.items():
    	if column not in new_data:
    		new_data[column] = {index: value}
    	else:
    		new_data[column][index] = value
```
?

Aparty from that I got no more comments and LGTM.